src/Main/ToolContext.h

```diff
@@ -39,11 +39,11 @@ class ToolContext {
         m_dataPath(DataPath) {}
   ~ToolContext() {}
 
-  const std::string& ToolName() { return m_toolName; }
-  const std::string& CompanyName() { return m_companyName; }
-  const std::string& ExecutableName() { return m_executableName; }
-  const std::filesystem::path& BinaryPath() { return m_binaryPath; }
-  const std::filesystem::path& DataPath() { return m_dataPath; }
+  const std::string& ToolName() const { return m_toolName; }
+  const std::string& CompanyName() const { return m_companyName; }
+  const std::string& ExecutableName() const { return m_executableName; }
+  const std::filesystem::path& BinaryPath() const { return m_binaryPath; }
+  const std::filesystem::path& DataPath() const { return m_dataPath; }
 
   void ToolName(const std::string& name) { m_toolName = name; }
```

Comment:
Why not SetToolName() ? bad code practice

Reply:
Google coding style prefers the getters and setters to be the same name:
 void ToolName(const std::string& name) { m_toolName = name; }
 const std::string& ToolName() { return m_toolName; }

Reply:
I didn't found this in google codding style. If you know where is it please send some link. I think it isn't a good style.

Reply:
I'm collaborating with Google employees on another project, that style rule came from there, it is not on their published style sheet but is it mandated or strongly suggested to use internally. Your style vs Google style :-)

Reply:
okey